File: vortexdbg-api/src/main/java/com/vortexdbg/unix/struct/SockAddr.java
package com.vortexdbg.unix.struct;

import com.vortexdbg.pointer.VortexdbgStructure;
import com.sun.jna.Pointer;

import java.util.Arrays;
import java.util.List;

public class SockAddr extends VortexdbgStructure {

    public SockAddr(Pointer p) {
        super(p);
    }

    public short sin_family;
    public short sin_port;
    public byte[] sin_addr = new byte[24];

    @Override
    protected List<String> getFieldOrder() {
        return Arrays.asList("sin_family", "sin_port", "sin_addr");
    }
}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/unix/struct/SockAddr.kt
package com.vortexdbg.unix.struct

import com.vortexdbg.pointer.VortexdbgStructure
import com.sun.jna.Pointer

class SockAddr(p: Pointer?) : VortexdbgStructure(p) {

    @JvmField
    var sin_family: Short = 0
    @JvmField
    var sin_port: Short = 0
    @JvmField
    var sin_addr: ByteArray = ByteArray(24)

    override fun getFieldOrder(): List<String> {
        return listOf("sin_family", "sin_port", "sin_addr")
    }
}
